src/main/java/de/espend/idea/php/generics/indexer/TemplateAnnotationIndex.java

```diff
@@ -103,4 +92,114 @@ public int getVersion() {
     public boolean dependsOnFileContent() {
         return true;
     }
+
+    private static class MyPsiRecursiveElementWalkingVisitor extends PsiRecursiveElementWalkingVisitor {
+        private final Map<String, TemplateAnnotationUsage> map;
+
+        public MyPsiRecursiveElementWalkingVisitor(Map<String, TemplateAnnotationUsage> map) {
+            this.map = map;
+        }
+
+        @Override
+        public void visitElement(@NotNull PsiElement element) {
+            if (element instanceof PhpClass) {
+                visitPhpClass((PhpClass) element);
+            } else if (element instanceof Function) {
+                visitPhpClass((Function) element);
+            }
+
+            super.visitElement(element);
+        }
+
+        private void visitPhpClass(PhpClass phpClass) {
+            String fqn = phpClass.getFQN();
+            if(fqn.startsWith("\\")) {
+                fqn = fqn.substring(1);
+            }
+
+            // doctrine has many tests: Doctrine\Tests\Common\Annotations\Fixtures
+            // we are on index process, project is not fully loaded here, so filter name based tests
+            // eg PhpUnitUtil.isTestClass not possible
+            if (!fqn.contains("\\Tests\\") && !fqn.contains("\\Fixtures\\") && GenericsUtil.isGenericsClass(phpClass)) {
+                map.put(fqn, new TemplateAnnotationUsage(fqn, TemplateAnnotationUsage.Type.CONSTRUCTOR, 0));
+            }
+        }
+
+        private void visitPhpClass(Function function) {
+            for (String docBlock : Arrays.asList("@template", "@psalm-template")) {
```

Comment:
Any plans to add also phpstan-* prefixed annotations? Most of them should be identical.
I may help prepare a compatibility list

Reply:
thank you, good point. i checked some github repos and it looks like they are also used a lot. i will centralized it